src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -6241,6 +6312,15 @@ void AuraEffect::HandlePeriodicHealAurasTick(Unit* target, Unit* caster) const
     TC_LOG_DEBUG("spells.periodic", "PeriodicTick: %s heal of %s for %u health inflicted by %u",
         GetCasterGUID().ToString().c_str(), target->GetGUID().ToString().c_str(), damage, GetId());
 
+    if (partial && m_period > 0)
+    {
+        int32 partialTickDuration = GetBase()->GetTickRemainingPeriod();
+        float dmgPercent = partialTickDuration / m_period;
```

Comment:
same as above (promote to float, then divide)

also, have you considered moving this to a helper method since it's always the same? (GetPartialTickPercent returning this value)